bindings/java/src/org/sleuthkit/datamodel/CaseDbAccessManager.java

```diff
@@ -675,5 +723,256 @@ private void validateSQL(String sql) throws TskCoreException {
 		 * TODO (JIRA-5950): Need SQL injection defense in CaseDbAccessManager 
 		 */
 	}
+	
+	
+	/**
+	 * A wrapper around a PreparedStatement to execute queries against the
+	 * database.
+	 */
+	@Beta
+	public class CaseDbPreparedStatement implements AutoCloseable {
+
+		private final CaseDbConnection connection;
+		private final PreparedStatement preparedStatement;
+		private final String originalSql;
+		private final boolean isWriteLockRequired;
```

Comment:
Tiny thing, but I think the code would be more readable if this was named something like "hasWriteLock" (keep the name for the parameter). You could set it immediately after getting the lock to make things very clear.